dms/models/ir_attachment.py

```diff
@@ -7,26 +7,24 @@ class IrAttachment(models.Model):
     _inherit = "ir.attachment"
 
     def _get_dms_directories(self, res_model, res_id):
-        return (
-            self.env["dms.directory"]
-            .sudo()
-            .search(
-                [
-                    ("res_model", "=", res_model),
-                    ("res_id", "=", res_id),
-                    ("storage_id.save_type", "=", "attachment"),
-                ]
-            )
+        return self.env["dms.directory"].search(
+            [
+                ("res_model", "=", res_model),
+                ("res_id", "=", res_id),
+                ("storage_id.save_type", "=", "attachment"),
+            ]
         )
 
     def _dms_directories_create(self):
         items = self._get_dms_directories(self.res_model, False)
         for item in items:
-            model_item = self.env[self.res_model].sudo().browse(self.res_id)
+            model_item = self.env[self.res_model].browse(self.res_id)
             ir_model_item = self.env["ir.model"].search(
                 [("model", "=", self.res_model)]
             )
-            self.env["dms.directory"].sudo().create(
+            self.env["dms.directory"].sudo().with_context(
+                bypass_check_name=True
```

Comment:
Following the other comment above:

```suggestion
                check_name=False
```